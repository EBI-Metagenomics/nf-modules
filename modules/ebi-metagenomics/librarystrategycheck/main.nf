
process LIBRARYSTRATEGYCHECK {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:1.2.1--pyhdfd78af_0':
        'biocontainers/mgnify-pipelines-toolkit:1.2.1--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(bcv)

    output:
    tuple val(meta), env(check_out), emit: library_check_out
    path "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    library_strategy_check \\
        -i ${bcv} \\
        -s ${prefix} \\
        -o ./

    check_out=\$(cat ${prefix}_library_check_out.txt)

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """

    stub:
    """
    check_out="AMPLICON"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """
}
