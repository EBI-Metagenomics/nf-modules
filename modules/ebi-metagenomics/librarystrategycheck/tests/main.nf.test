
nextflow_process {

    name "Test Process LIBRARYSTRATEGYCHECK"
    script "../main.nf"
    process "LIBRARYSTRATEGYCHECK"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "librarystrategycheck"

    test("librarystrategycheck - AMPLICON") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false], // meta map
                    file("${moduleDir}/tests/fixtures/test_bcv.tsv", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.library_check_out,
                    process.out.versions.collect { path(it).yaml }
                    ).match()
                }
            )
        }
    }

    test("librarystrategycheck - NOT AMPLICON") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false], // meta map
                    file("${moduleDir}/tests/fixtures/test_bcv_notamplicon.tsv", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.library_check_out,
                    process.out.versions.collect { path(it).yaml }
                    ).match()
                }
            )
        }
    }

    test("librarystrategycheck - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false], // meta map
                    file("${moduleDir}/tests/fixtures/test_bcv.tsv", checkIfExists: true)
                    ]
                """
            }
            }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.library_check_out,
                    process.out.versions.collect { path(it).yaml }
                    ).match()
                }
            )
        }
    }
}
