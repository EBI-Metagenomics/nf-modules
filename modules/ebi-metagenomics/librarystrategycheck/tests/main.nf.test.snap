{
    "librarystrategycheck - NOT AMPLICON": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    ""
                ]
            ],
            [
                {
                    "LIBRARYSTRATEGYCHECK": {
                        "mgnify-pipelines-toolkit": "1.2.1"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-18T14:42:44.234847"
    },
    "librarystrategycheck - AMPLICON": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "AMPLICON"
                ]
            ],
            [
                {
                    "LIBRARYSTRATEGYCHECK": {
                        "mgnify-pipelines-toolkit": "1.2.1"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-18T14:42:35.510857"
    },
    "librarystrategycheck - stub": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "AMPLICON"
                ]
            ],
            [
                {
                    "LIBRARYSTRATEGYCHECK": {
                        "mgnify-pipelines-toolkit": "1.2.1"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-18T14:42:51.964961"
    }
}